trl/trainer/judges.py

```diff
@@ -134,6 +134,46 @@ def judge(self, prompts: List[str], completions: List[List[str]], shuffle_order:
         raise NotImplementedError("Judge subclasses must implement the `judge` method.")
 
 
+class BaseConstraintJudge(BaseJudge):
```

Comment:
I'm trying to think wider than the paper. What could be a better name for such a judge? (ie a judge that returns 0/1)? `BinaryJudge`?

Reply:
yes in the nash-md terminology its a "hard-judge"

Reply:
soft-judge is when the judge returns a value in [0, 1]?

Reply:
> I'm trying to think wider than the paper. What could be a better name for such a judge? (ie a judge that returns 0/1)? `BinaryJudge`?

Yeah, that's how I named it in the first commits.